nni/retiarii/oneshot/pytorch/base_lightning.py

```diff
@@ -0,0 +1,223 @@
+        Combine architecture optimizers and user's model optimizers.
+        Overwrite configure_architecture_optimizers if architecture optimizers
+        are needed in your NAS algorithm.
+        By default ''self.model'' is currently a _SupervisedLearningModule in
+        nni.retiarii.evaluator.pytorch.lightning, and it only returns 1 optimizer.
+        But for extendibility, codes for other return value types are also implemented.
+        """
+        arc_optimizers = self.configure_architecture_optimizers()
+        if arc_optimizers is None:
+            return self.model.configure_optimizers()
+
+        if isinstance(arc_optimizers, optim.Optimizer):
+            arc_optimizers = [arc_optimizers]
+        self.arc_optim_count = len(arc_optimizers)
+
+        # The third return value ''frequency'' and ''monitor'' are ignored since lightning
+        # requires len(optimizers) == len(frequency), and gradient backword
+        # is handled manually.
+        w_optimizers, lr_schedulers, _, _ = \
+            self.trainer._configure_optimizers(self.model.configure_optimizers())
+
+        return arc_optimizers + w_optimizers, lr_schedulers
+
+    def on_train_start(self):
+        return self.model.on_train_start()
+
+    def on_train_end(self):
+        return self.model.on_train_end()
+
+    def on_fit_start(self):
+        return self.model.on_train_start()
+
+    def on_fit_end(self):
+        return self.model.on_train_end()
+
+    def on_train_batch_start(self, batch, batch_idx, unused = 0):
+        return self.model.on_train_batch_start(batch, batch_idx, unused)
+
+    def on_train_batch_end(self, outputs, batch, batch_idx, unused = 0):
+        return self.model.on_train_batch_end(outputs, batch, batch_idx, unused)
+
+    def on_epoch_start(self):
+        return self.model.on_epoch_start()
+
+    def on_epoch_end(self):
+        return self.model.on_epoch_end()
+
+    def on_train_epoch_start(self):
+        return self.model.on_train_epoch_start()
+
+    def on_train_epoch_end(self):
+        return self.model.on_train_epoch_end()
+
+    def on_before_backward(self, loss):
+        return self.model.on_before_backward(loss)
+
+    def on_after_backward(self):
+        return self.model.on_after_backward()
+
+    def configure_gradient_clipping(self, optimizer, optimizer_idx, gradient_clip_val = None, gradient_clip_algorithm = None):
+        return self.model.configure_gradient_clipping(optimizer, optimizer_idx, gradient_clip_val, gradient_clip_algorithm)
+
+    def configure_architecture_optimizers(self):
+        '''
+        Hook kept for subclasses. Each specific NAS method returns the optimizer of
+        architecture parameters.
+
+        Returns
+        ----------
+        arc_optimizers : List[Optimizer], Optimizer
+            Optimizers used by a specific NAS algorithm for its architecture parameters.
+            Return None if no architecture optimizers are needed.
+        '''
+        arc_optimizers = None
```

Comment:
return None?

Reply:
There is no specific optimizers in base class.

Reply:
I mean,

```python
arc_optimizers = None
return arc_optimizers
```

is equivalent to,

```python
return None
```